### ISSUE TITLE
promise and js-interop dependencies not wanted

### ISSUE THREAD

**Issue by henryw374** [lambdaisland/fetch#18]
Hi - thanks for the lib!

I removed those dependencies in my fork https://github.com/henryw374/fetch/commit/080f084ea177060489df5b8a3bd93c093c3fa220 and it seems to work ok  -incl. with advanced compile. 

Without any of my changes the tests fail though - complains about kaocha not found. 

If wanted, and if tests would pass, I could do more tidy to make PR

**Comment by plexus** [comment]
What's your motivation for removing these? What problem does this solve?

Perhaps this affects build size, but since these are all macros that translate to raw interop I would expect any change to be small. Have you done any measurements on that?

Unless there's a pretty big payoff I'd prefer to keep the version we have. These are the interop libs we use across projects, and I consider the code as it is more easily maintainable than the raw interop version.

**Comment by henryw374** [comment]
It's just the fact that we use other techniques/libraries for interop/promises generally - so I'd rather not have extra stuff on the classpath that will inevitably creep in to our code.

The changes needed to remove those deps were pretty small. IOW it didn't feel like they were really making enough of a difference to justify having them there.

**Comment by geraldodev** [comment]
It would be nice a version of this libraries with minimum dependencies, I've started a fork for myself because I don't want to use core.async. I love applied-science interop but I see that it possible that some ppl does not want to use it. I've changed kitchen-async to promesa library, because it's better documented, and does depend on core.async. Folks are using ^js annotation to mark bits of code that must survive advanced compilation, as far as I can remember.
